File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v3/UtaInsLoanApi.java
package com.bitget.openapi.api.v3;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface UtaInsLoanApi {

    @GET("/api/v3/ins-loan/product-infos")
    Call<ResponseResult> productInfos(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/ins-loan/ensure-coins-convert")
    Call<ResponseResult> ensureCoinsConvert(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/ins-loan/symbols")
    Call<ResponseResult> symbols(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/ins-loan/loan-order")
    Call<ResponseResult> loanOrder(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/ins-loan/repaid-history")
    Call<ResponseResult> repaidHistory(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/ins-loan/ltv-convert")
    Call<ResponseResult> ltvConvert(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/ins-loan/transfered")
    Call<ResponseResult> transfered(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/ins-loan/risk-unit")
    Call<ResponseResult> riskUnit();

    @POST("/api/v3/ins-loan/bind-uid")
    Call<ResponseResult> bindUid(@Body Map<String, Object> paramMap);
}

File: bitget-node-sdk-api/src/lib/v3/UtaInsLoanApi.ts
import {BaseApi} from '../BaseApi';

export class UtaInsLoanApi extends BaseApi {

    // GET /api/v3/ins-loan/product-infos
    productInfo(qsOrBody: object) {
        const url = '/api/v3/ins-loan/product-infos';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/ins-loan/ensure-coins-convert
    ensureCoinsConvert(qsOrBody: object) {
        const url = '/api/v3/ins-loan/ensure-coins-convert';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/ins-loan/symbols
    symbols(qsOrBody: object) {
        const url = '/api/v3/ins-loan/symbols';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/ins-loan/loan-order
    loanOrder(qsOrBody: object) {
        const url = '/api/v3/ins-loan/loan-order';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/ins-loan/repaid-history
    repayHistory(qsOrBody: object) {
        const url = '/api/v3/ins-loan/repaid-history';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/ins-loan/ltv-convert
    ltvConvert(qsOrBody: object) {
        const url = '/api/v3/ins-loan/ltv-convert';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/ins-loan/transfered
    transferAsset(qsOrBody: object) {
        const url = '/api/v3/ins-loan/transfered';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/ins-loan/risk-unit
    riskUnit(qsOrBody: object) {
        const url = '/api/v3/ins-loan/risk-unit';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/ins-loan/bind-uid
    bindUid(qsOrBody: object) {
        const url = '/api/v3/ins-loan/bind-uid';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }
}
